src/main/java/com/shaft/api/RestActions.java

```diff
@@ -523,116 +253,376 @@ public static String getResponseXMLValue(Object response, String xmlPath) {
+     * @param comparisonType        ComparisonType.EQUALS, CONTAINS; Note that
+     *                              MATCHES ignores the content ordering inside the
+     *                              JSON
+     * @param jsonPathToTargetArray a jsonpath that will be parsed to point to the
+     *                              target JSON Array
+     * @return a boolean value that is TRUE in case the comparison passed, or FALSE
+     * in case it failed
+     */
+    public static boolean compareJSON(Response response, String referenceJsonFilePath, ComparisonType comparisonType,
+                                      String jsonPathToTargetArray) {
+        if (jsonPathToTargetArray.equals("")) {
+            ReportManager.logDiscrete("Comparing the provided API response with the file at this path \""
+                    + referenceJsonFilePath + "\", comparison type \"" + comparisonType + "\"");
+        } else {
+            ReportManager.logDiscrete("Comparing the provided API response with the file at this path \""
+                    + referenceJsonFilePath + "\", comparison type \"" + comparisonType
+                    + "\", jsonPath to target array \"" + jsonPathToTargetArray + "\".");
+        }
+        boolean comparisonResult;
+        JSONParser parser = new JSONParser();
+        List<Object> expectedJSONAttachment = null;
+
+        try {
+            // parse actual JSON into Object or Array
+            org.json.simple.JSONObject actualJsonObject = null;
+            org.json.simple.JSONArray actualJsonArray = null;
+
+            var actualObject = parser.parse(response.asString());
+            if (actualObject instanceof org.json.simple.JSONObject) {
+                actualJsonObject = (org.json.simple.JSONObject) parser.parse(response.asString());
+            } else {
+                // actualObject is an array org.json.simple.JSONArray
+                actualJsonArray = (org.json.simple.JSONArray) parser.parse(response.asString());
+            }
+
+            // parse expected JSON into Object or Array
+            org.json.simple.JSONObject expectedJsonObject = null;
+            org.json.simple.JSONArray expectedJsonArray = null;
+
+            var expectedObject = parser.parse(new FileReader(referenceJsonFilePath));
+            if (expectedObject instanceof org.json.simple.JSONObject) {
+                expectedJsonObject = (org.json.simple.JSONObject) parser.parse(new FileReader(referenceJsonFilePath));
+                expectedJSONAttachment = Arrays.asList("File Content", "Expected JSON",
+                        new GsonBuilder().setPrettyPrinting().create()
+                                .toJson(new JsonParser().parse(expectedJsonObject.toJSONString())));
+            } else {
+                // expectedObject is an array org.json.simple.JSONArray
+                expectedJsonArray = (org.json.simple.JSONArray) parser.parse(new FileReader(referenceJsonFilePath));
```

Comment:
Codacy found an issue: [Unnecessary use of fully qualified name 'org.json.simple.JSONArray' due to existing import 'org.json.simple.JSONArray'](https://app.codacy.com/manual/mohab.mohieeldeen/SHAFT_ENGINE/pullRequest?prid=5737302)